### PR TITLE
feat/create grouped investigations on button click

### PR DESCRIPTION
**Pull request by ItayBenMoshe**
[848](https://dev.azure.com/spectrumFactory/CoronaI/_workitems/edit/848/)

### CODE REVIEW

File: server/src/ClientToDBAPI/GroupedInvestigationsRoute/mainRoute.ts [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+            const invetigationsToGroupMessage = invetigationsToGroup.join(', ');
+            groupedInvestigationsLogger.info(`updating investigations ${invetigationsToGroupMessage} to be part of group ${groupToCreate.id}`, Severity.LOW);
+            graphqlRequest(UPDATE_GROUPED_INVESTIGATIONS, response.locals, {
+                epidemiologyNumbers: invetigationsToGroup,
+                groupId: groupToCreate.id
+            })
+            .then(result => {
+                if (result?.data?.updateGroupedInvestigations) {
+                    groupedInvestigationsLogger.info(`investigations ${invetigationsToGroupMessage} were updated successfully`, Severity.LOW);
+                    console.log(result);
```
Comment (nevobonneh) [inline server/src/ClientToDBAPI/GroupedInvestigationsRoute/mainRoute.ts:62]: delete console.log

Comment (ItayBenMoshe) [reply server/src/ClientToDBAPI/GroupedInvestigationsRoute/mainRoute.ts:62]: fixed

**Review by nevobonneh — DISMISSED** [review]

File: client/src/components/App/Content/LandingPage/InvestigationTable/InvestigationTableFooter/GroupedInvestigations/useGroupedInvestigations.ts [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+        const invetigationsToGroupIds = invetigationsToGroup.map((invetigationToGroup: InvestigationTableRow) => invetigationToGroup.epidemiologyNumber);
+        setIsLoading(true);
+        axios.post('/groupedInvestigations', { groupToCreate, invetigationsToGroupIds })
+        .then(() => {
+            setIsLoading(false);
+            onClose();
+            groupToCreateLogger.info('create grouped investigations successfully', Severity.LOW);
+        })
+        .catch((err) => {
+            groupToCreateLogger.error(`create grouped investigations was failde due to${err}`, Severity.HIGH);
```
Comment (RoeiRom) [inline client/src/components/App/Content/LandingPage/InvestigationTable/InvestigationTableFooter/GroupedInvestigations/useGroupedInvestigations.ts:49]: move setIsLoading to finally

File: server/src/ClientToDBAPI/GroupedInvestigationsRoute/mainRoute.ts [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+groupedInvestigationsRoute.post('/', adminMiddleWare, (request: Request, response: Response) => {
+    const groupToCreateLogger = logger.setup({
+        workflow: 'create group for investigations',
+        user: response.locals.user.id,
+        investigation: response.locals.epidemiologynumber
+    })
+    const groupToCreate = request.body.groupToCreate; 
+    groupToCreateLogger.info(`launching create group info with the parameters ${JSON.stringify(groupToCreate)}`, Severity.LOW);
+    graphqlRequest(CREATE_GROUP_FOR_INVESTIGATIONS, response.locals, {input: groupToCreate})
```
Comment (RoeiRom) [inline server/src/ClientToDBAPI/GroupedInvestigationsRoute/mainRoute.ts:43]: why not uniting both create and update to one request?

File: client/src/components/App/Content/LandingPage/InvestigationTable/InvestigationTableFooter/GroupedInvestigations/useGroupedInvestigations.ts (hunk 2 above)
Comment (nevobonneh) [reply client/src/components/App/Content/LandingPage/InvestigationTable/InvestigationTableFooter/GroupedInvestigations/useGroupedInvestigations.ts:49]: failde

File: server/src/ClientToDBAPI/GroupedInvestigationsRoute/mainRoute.ts (hunk 3 above)
Comment (ItayBenMoshe) [reply server/src/ClientToDBAPI/GroupedInvestigationsRoute/mainRoute.ts:43]: what do you mean?

Comment (RoeiRom) [reply server/src/ClientToDBAPI/GroupedInvestigationsRoute/mainRoute.ts:43]: here you 2 requests in order to create the group and populate it, why not populating the investigation when it is created

File: server/src/DBService/GroupedInvestigations/Query.ts [hunk 4]
```diff
@@ -0,0 +1,12 @@
+import { gql } from 'postgraphile';
+
+export const GET_GROUPED_INVESTIGATIONS_REASONS = gql`
+query GetAllGroupedInvestigationsReasons {
+  allInvestigationGroupReasons {
+    nodes {
+    	id
+			displayName
```
Comment (EmilyLip) [inline server/src/DBService/GroupedInvestigations/Query.ts:8]: remove space

File: server/src/ClientToDBAPI/GroupedInvestigationsRoute/mainRoute.ts [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+    });
+});
+
+groupedInvestigationsRoute.post('/', adminMiddleWare, (request: Request, response: Response) => {
+    const groupToCreateLogger = logger.setup({
+        workflow: 'create group for investigations',
+        user: response.locals.user.id,
+        investigation: response.locals.epidemiologynumber
+    })
+    const groupToCreate = request.body.groupToCreate;
```
Comment (EmilyLip) [inline server/src/ClientToDBAPI/GroupedInvestigationsRoute/mainRoute.ts:41]: maybe change to newGroup

File: server/src/ClientToDBAPI/GroupedInvestigationsRoute/mainRoute.ts [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+        reasonsLogger.error(`failed to fetch reasons due to ${error}`, Severity.HIGH);
+        response.status(errorStatusCode).json({ error: 'failed to fetch reasons' });
+    });
+});
+
+groupedInvestigationsRoute.post('/', adminMiddleWare, (request: Request, response: Response) => {
+    const groupToCreateLogger = logger.setup({
+        workflow: 'create group for investigations',
+        user: response.locals.user.id,
+        investigation: response.locals.epidemiologynumber
```
Comment (EmilyLip) [inline server/src/ClientToDBAPI/GroupedInvestigationsRoute/mainRoute.ts:39]: add the investigations we want to group in the log messaage

File: server/src/DBService/GroupedInvestigations/Query.ts (hunk 4 above)
Comment (ItayBenMoshe) [reply server/src/DBService/GroupedInvestigations/Query.ts:8]: weird. I can't see it in my file

File: server/src/ClientToDBAPI/GroupedInvestigationsRoute/mainRoute.ts (hunk 6 above)
Comment (ItayBenMoshe) [reply server/src/ClientToDBAPI/GroupedInvestigationsRoute/mainRoute.ts:39]: but here i just create the group

File: server/src/ClientToDBAPI/GroupedInvestigationsRoute/mainRoute.ts (hunk 5 above)
Comment (ItayBenMoshe) [reply server/src/ClientToDBAPI/GroupedInvestigationsRoute/mainRoute.ts:41]: same meaning, don't you think?

File: server/src/ClientToDBAPI/GroupedInvestigationsRoute/mainRoute.ts (hunk 6 above)
Comment (EmilyLip) [reply server/src/ClientToDBAPI/GroupedInvestigationsRoute/mainRoute.ts:39]: It does not have to be exactly here, the goal is to have all of the inforamtion about the creation of the group written in the logs (reason, epidemiological numbers, etc.)

File: server/src/ClientToDBAPI/GroupedInvestigationsRoute/mainRoute.ts (hunk 5 above)
Comment (EmilyLip) [reply server/src/ClientToDBAPI/GroupedInvestigationsRoute/mainRoute.ts:41]: agreed,, but i think newGroup represents it better 
not critical

Comment (ItayBenMoshe) [reply server/src/ClientToDBAPI/GroupedInvestigationsRoute/mainRoute.ts:41]: so i keep it groupToCreate. (it is groupToCreate in my schmea also)

**Review by RoeiRom — APPROVED** [review]

**Review by EmilyLip — APPROVED** [review]

**Pull request merged**